website/project/model.py

```diff
@@ -852,15 +855,43 @@ class Node(GuidStoredObject, AddonModelMixin, IdentifierMixin):
     # Primary institution node is attached to
     primary_institution = fields.ForeignField('institution')
 
+    affiliated_institutions = fields.ForeignField('institution', list=True)
+
     def add_primary_institution(self, user, inst):
         if inst.auth(user):
             self.primary_institution = inst
+            self.affiliated_institutions.append(inst)
             self.save()
+            self.add_log(
+                action=NodeLog.INSTITUTION_ADDED,
+                params={
+                    'node': self._primary_key,
+                    'institution': {
+                        'id': inst._id,
+                        'name': inst.name
+                    }
+                },
+                auth=Auth(user)
+            )
             return True
         return False
 
-    def remove_primary_institution(self):
+    def remove_primary_institution(self, user):
+        inst = self.primary_institution
         self.primary_institution = None
+        if inst in self.affiliated_institutions:
+            self.affiliated_institutions.remove(inst)
+        self.add_log(
```

Comment:
A log should only be added if the institution was actually removed.

If the institution isn't affiliated with the user, then an exception should probably be raised.